### PR TITLE
[_]: Fix/checkout id not being retrieved

### PR DESCRIPTION
**Pull request by fabioespinosa**
- Fixes sessionId not appearing in analytics

### CODE REVIEW

**Comment by fabioespinosa** [comment]
@JoanMora

File: src/app/payment/views/CheckoutSuccessView/CheckoutSuccessView.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const CheckoutSuccessView = (): JSX.Element => {
 
   useEffectAsync(async () => {
     await dispatch(userThunks.refreshUserThunk());
-    analyticsService.trackPaymentConversion();
+    await analyticsService.trackPaymentConversion();
```
Comment (sg-gs) [inline src/app/payment/views/CheckoutSuccessView/CheckoutSuccessView.tsx]: do not make the checkout depend on analytics

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/app/payment/views/CheckoutSuccessView/CheckoutSuccessView.tsx (hunk 1 above)
Comment (fabioespinosa) [reply src/app/payment/views/CheckoutSuccessView/CheckoutSuccessView.tsx]: it has a try catch inside though 🤔

Comment (sg-gs) [reply src/app/payment/views/CheckoutSuccessView/CheckoutSuccessView.tsx]: It would be better not to rely on service as the component is responsible for its own behavior. So if the component has to handle the checkout success and sending the analytics is not critical for handling the success is the component who is responsible for ensuring that behavior really happens as expected. 

Therefore, if someone decides to remove the `try-catch` from the analytics, all the App will not be broken, and the side effects are avoided. 

'The checkout is suddenly broken'. Then you find that someone removed something non-directly related to the checkout but still break it, which is very serious, troublesome to debug and a nightmare that always returns to you.

In summary, a good idea will be to extract this function from the hook if possible and add a `try-catch` there to ensure that the component accomplishes its responsibilities always. Then the dependency from the analytics is avoided.

**Review by sg-gs — APPROVED** [review]

**Review by JoanMora — COMMENTED** [review]
Can we send an error here?
window.rudderanalytics.track('Payment Conversion Error', {
msg: string
})
